docs/proposals/policy.md

```diff
@@ -0,0 +1,264 @@
+  1.  get
+  1.  create
+  1.  update
+  1.  delete
+  1.  proxy
+  1.  * - represents all verbs
+
+### Verb Specific Attributes
+Different verbs will require different attributes.  For instance, a list would have a kind, a get would have an kind and an id, and an update may have a kind, an id, and a list of the fields being modified.  Although exec isn't well defined, it could very well have an entirely distinct set of attributes.  The exact contract rules will be chosen by the Authorizer/AuthorizationAttributeBuilder pair, but basic resource access for openshift should include kind in order to be compatible with existing kubernetes authorization.
+
+#### Why allow TeamNames instead of simply saying that RoleBindings only have a list of UserNames?  
+Ease of maintenance and separation of powers.  Teams span namespaces, but RoleBindings do not.  If you specify the full set of users on every RoleBinding, then you have to repeat it for every namespace.  If that membership changes, you now have to update it in multiple locations.  As for the separation of powers, if you have rights to modify a RoleBinding, you have the power to change its permissions (RoleTemplateName) instead of only having the power to change its membership.  It seems reasonable to allow someone to modify team membership ("chubba is mechanic") without also having the ability to modify permissions associated with the team itself ("mechanics are super-admins").  Having Teams distinct from Roles and RoleBindings makes it easy to express this in policy.
+
+## API
+### Basic CRUD
+The basic CRUD for Roles and RoleBindings, and Policy is not as straightforward as usual.  
+
+The REST API will respond to gets, lists, creates, updates, and deletes (not watches) at the expected locations for Roles and RoleBindings.  The REST API will respond to gets, lists, and watches (not creates, updates, and deletes) at the expected location for Policy.  
+
+Only one Policy document is allowed per namespace.  Roles and RoleBindings are not stored in etcd.  Instead, requests against their REST endpoints will result in inspection and modification of the Policy document.  This makes it easier to express restrictions on how policy can be manipulated, while making it possible to guarantee of referential integrity for references inside the local namespace.
+
+#### allowedentities?namespace=<namespace>
+**Please suggest a better name and a better location to put this** Given a namespace which users and teams can see some resource contained in that namespace.  This API is not guaranteed to accurate results.  It is possible that a user returned in the list, could be denied access to any resource in the namespace.  This can happen when a user is granted allow rights for something the namespace, but a deny rule later stomps on it.  Barring weird cases like that, it will work.  The response will look like this:
+```
+{
+	"users": ["someuser"],
+	"teams": ["someteam"]
+}
+```
+
+#### cando?request=<request>
```

Comment:
I thought this would be used to evaluate policy for a specified user, not for the caller (for example, by a proxy deciding whether to run a command from a user on a container)